### ISSUE TITLE
Adaptive learning rate

### ISSUE THREAD

**Issue by peastman** [deepchem/deepchem#2547]
DeepChem offers various learning rate schedules, but all of them require you to specify the full schedule in advance.  A popular alternative is to monitor the loss and reduce the learning rate whenever it plateaus.  This approach generally gives reasonable training behavior while requiring less manual tuning of hyperparameters by the user.  We should consider adding it.

**Comment by rbharath** [comment]
This would be a nice addition! I've done something similar by writing a manual training loop but it would be nice to have a better API

**Comment by atreyamaj** [comment]
Hi! So I was just looking through torch and tf documentation, maybe we can integrate the following as a temporary solution and add more later:

[ReduceLRonPlateau Tensorflow/Keras](https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/ReduceLROnPlateau)

[ReduceLRonPlateau Torch](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ReduceLROnPlateau.html)
